emperor/support_files/js/view-controller.js

```diff
@@ -253,6 +239,75 @@ define([
      * application. This argument is passed by reference.
      */
     this.decompViewDict = decompViewDict;
+
+    return this;
+  }
+  EmperorViewController.prototype = Object.create(
+      EmperorViewControllerABC.prototype);
+  EmperorViewController.prototype.constructor = EmperorViewControllerABC;
+
+  /**
+   * Retrieves the metadata field currently being controlled
+   *
+   * @return {String} A key corresponding to the active decomposition view.
+   */
+  EmperorViewController.prototype.getActiveDecompViewKey = function() {
+    return this.activeViewKey;
+  };
+
+  /**
+   * Changes the metadata column name to control.
+   *
+   * @param {String} k Key corresponding to active decomposition view.
+   */
+  EmperorViewController.prototype.setActiveDecompViewKey = function(k) {
+    // FIXME: this should be validated against decompViewDict i.e. we should be
```

Comment:
sorry, this comment got placed in the wrong place, meant i can look through as a second reviewer